controllers/front/CartController.php

```diff
@@ -326,12 +350,13 @@ protected function processChangeProductInCart()
         }
 
         // Check product quantity availability
-        if ($this->id_product_attribute) {
-            if (!Product::isAvailableWhenOutOfStock($product->out_of_stock) && !Attribute::checkAttributeQty($this->id_product_attribute, $qty_to_check)) {
-                $this->errors[] = $this->trans('There are not enough products in stock', array(), 'Shop.Notifications.Error');
-            }
-        } elseif (!$product->checkQty($qty_to_check)) {
-            $this->errors[] = $this->trans('There are not enough products in stock', array(), 'Shop.Notifications.Error');
+        $ErrorKey = $isUpdateOperationDown ? 'updateOperationError' : 'errors';
+        $hasQuantityError = ($this->id_product_attribute) ?
+            (!Product::isAvailableWhenOutOfStock($product->out_of_stock) && !Attribute::checkAttributeQty($this->id_product_attribute, $qty_to_check))
+            : (!$product->checkQty($qty_to_check && !$isUpdateOperationDown));
```

Comment:
Ternary operators are cool for short tests and short assignations.
But here this is really difficult to read. Maybe an if/else is better.